include/envoy/event/dispatcher.h

```diff
@@ -90,7 +90,7 @@ class Dispatcher {
    *               initially listen on.
    */
   virtual FileEventPtr createFileEvent(int fd, FileReadyCb cb, FileTriggerType trigger,
-                                       uint32_t events) PURE;
+                                       uint32_t events) const PURE;
```

Comment:
We are somewhat splitting hairs here, but IMO this is not a const operation on the dispatcher, it does modify its state. Instead of changing this to a const method can you pass a non-const Dispatcher to where you need it? This is what is done elsewhere.